src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -99,150 +102,180 @@ public static StashPullRequest extractPullRequest(String project, String reposit
-
-                        // Add comment attached to the current line
-                        JSONArray jsonCommentIds = (JSONArray) jsonLine.get("commentIds");
-                        if (jsonCommentIds != null) {
-
-                          for (Object objCommentId: jsonCommentIds.toArray()) {
-                            long commentId = (long) objCommentId;
-
-                            JSONArray jsonLineComments = (JSONArray) jsonDiff.get("lineComments");
-                            if (jsonLineComments != null) {
-                              for (Object objLineComment : jsonLineComments.toArray()) {
-                                JSONObject jsonLineComment = (JSONObject) objLineComment;
-
-                                long lineCommentId = (long) jsonLineComment.get("id");
-                                if (lineCommentId == commentId) {
-
-                                  String lineCommentMessage = (String) jsonLineComment.get("text");
-                                  long lineCommentVersion = (long) jsonLineComment.get(VERSION);
-
-                                  JSONObject objAuthor = (JSONObject) jsonLineComment.get(AUTHOR);
-                                  if (objAuthor != null) {
-
-                                    StashUser author = extractUser(objAuthor);
-
-                                    StashComment comment = new StashComment(lineCommentId, lineCommentMessage, path, destination, author, lineCommentVersion);
-                                    diff.addComment(comment);
-
-                                    // get the tasks linked to the current comment
-                                    JSONArray jsonTasks = (JSONArray) jsonLineComment.get("tasks");
-                                    if (jsonTasks != null) {
-                                      for (Object objTask : jsonTasks.toArray()) {
-                                        JSONObject jsonTask = (JSONObject) objTask;
-
-                                        comment.addTask(extractTask(jsonTask.toString()));
-                                      }
-                                    }
-                                  }
-                                }
-                              }
-                            }
-                          }
-                        }
-
-                        result.add(diff);
-                      }
-                    }
-                  }
-                }
-              }
-            }
+    if (jsonDiffs == null) { return null; }
+
+    // Let's call this for loop "objdiff_loop"
+    for (Object objDiff : jsonDiffs.toArray()) {
+
+      JSONObject jsonDiff = (JSONObject)objDiff;
+      // destination path in diff view
+      // if status of the file is deleted, destination == null
+      JSONObject destinationPath = (JSONObject)jsonDiff.get("destination");
+
+      if (destinationPath == null) { continue; } // Let's process the next item in "objdiff_loop"

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)